pagopa/io-app
# feat: [IOBP-801] Add transactions category tab filters

## Short description
This PR implements category filters for the receipt list, allowing users to view receipts "paid by me," "addressed to me," or both. The selected filter is applied directly to the backend request, enhancing result customization. Additionally, this PR improves list performance by leveraging React memoization, reducing unnecessary re-renders and significantly enhancing overall efficiency.

## List of changes proposed in this pull request
- Added a new component `PaymentsBizEventsFilterTabs` that handles the tabs filter;
- Added `noticeCategory` attribute to the `getPaymentsBizEventsTransactionsAction` that can be "all", "payer" or "debtor". Based on the type of the value it will send to the backend request `is_debtor` or `is_payer` flag;
- Wrapped the `PaymentsBizEventsListItemTransaction` component inside a `React.memo` to avoid unnecessary re-rendering and allow the re-rendering only if the new attributes are different.

## How to test
- Open the "Payments" screen;
- Tap on the "See all" CTA on the top-right of the latest transactions to open the transactions list;
- You can see three tabs to filter the elements backend-side;
- If you're testing it on dev-server please check that the request populate the right query parameter;
- If you're testing it on UAT env. you can see the right result returned in the list;

## Preview

https://github.com/user-attachments/assets/26007dce-1631-48cb-8ce8-f1220eeca6aa
